### PR TITLE
feat: add video support for webstories

### PR DESCRIPTION
**Pull request by cricket-lover**
Reference: https://github.com/quintype/ace-planning/issues/676

### CODE REVIEW

File: src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
 import get from "lodash/get";
+import { VideoWebStory } from "../../video-web-story";
 
 const WebStoryPageComponentsBase = ({ card, config, story }: WebStoryPageComponentsTypes) => {
   const titleElement = card["story-elements"].find((el) => el.type === "title");
   const textElements = card["story-elements"].filter((el) => el.type === "text" && el.subtype !== "cta");
   const imageElement = card["story-elements"].find((el) => el.type === "image");
+
+  const videoElement = card["story-elements"].find((el) => el.type === "jsembed");
+  const videoUrl = videoElement && atob(`${videoElement["embed-js"]}`);
```
Comment (ags1773) [inline src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx:18]: move decoding to inside `VideoWebStory` component

File: src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx [hunk 2]
```diff
@@ -24,7 +30,8 @@ const WebStoryPageComponentsBase = ({ card, config, story }: WebStoryPageCompone
 
   return (
     <Fragment>
-      {imageElement && (
+      {videoUrl && <VideoWebStory videoUrl={videoUrl} config={config} imageElement={imageElement} />}
```
Comment (ags1773) [inline src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx:33]: no need to pass config this way. `VideoWebStory` component can get config from context

File: src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx [hunk 3]
```diff
@@ -24,7 +30,8 @@ const WebStoryPageComponentsBase = ({ card, config, story }: WebStoryPageCompone
...
+      {imageElement && !videoElement && (
```
Comment (ags1773) [inline src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx:34]: Also add the `!videoElement` condition below > `{(titleElement || textElements.length || imageElement)`
we don't want to show title/text elements on top of video
But CTA button should show

File: src/atoms/video-web-story/video-web-story.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+export const VideoWebStory = ({ videoUrl, config, imageElement }) => {
+  const poster = imageElement ? `https://${config.publisherConfig["cdn-image"]}/${imageElement["image-s3-key"]}` : null;
+  return (
+    <Fragment>
+      <Helmet>
+        <script async={undefined} custom-element="amp-video" src="https://cdn.ampproject.org/v0/amp-video-0.1.js" />
+      </Helmet>
+      <amp-story-grid-layer template="fill">
+        <amp-video autoplay="" poster={poster} width="640" height="360" layout="responsive">
```
Comment (ags1773) [inline src/atoms/video-web-story/video-web-story.tsx:12]: why hardcoding width & height?

File: src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx [hunk 5]
```diff
@@ -1,16 +1,22 @@
...
 import { getAnimationProps } from "./web-story-page-components.helpers";
 import get from "lodash/get";
+import { VideoWebStory } from "../../video-web-story";
 
 const WebStoryPageComponentsBase = ({ card, config, story }: WebStoryPageComponentsTypes) => {
   const titleElement = card["story-elements"].find((el) => el.type === "title");
   const textElements = card["story-elements"].filter((el) => el.type === "text" && el.subtype !== "cta");
   const imageElement = card["story-elements"].find((el) => el.type === "image");
+
+  const videoElement = card["story-elements"].find((el) => el.type === "jsembed");
```
Comment (ags1773) [inline src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx:17]: https://github.com/quintype/ace-planning/issues/676
```
- we'll parse the jsembed and see if it has a .mp4 or other such supported video extension (check which all videos are supported)
- if it is a supported video, we need to add it as a video in the web story template
```

We can't assume jsembed == video. There needs to be a check

File: src/atoms/video-web-story/video-web-story.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+          <source src={videoUrl} type="video/mp4" />
```
Comment (ags1773) [inline src/atoms/video-web-story/video-web-story.tsx:13]: what about other video formats?

**Review by ags1773 — CHANGES_REQUESTED** [review]

File: src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx (hunk 5 above)
Comment (cricket-lover) [reply src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx:16]: We already checked this in the canTakeCard.

File: src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx (hunk 3 above)
Comment (cricket-lover) [reply src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx:32]: done

File: src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx (hunk 5 above)
Comment (ags1773) [reply src/atoms/visual-story/web-story-page-components/web-story-page-components.tsx:16]: missed that. Looks ok

**Review by ags1773 — APPROVED** [review]

**Pull request merged**